docs/pages/application-access/getting-started.mdx

```diff
@@ -22,14 +22,21 @@ Let's connect to Grafana using Teleport in three steps:
 (!docs/pages/includes/edition-prereqs-tabs.mdx!)
 
 - (!docs/pages/includes/tctl.mdx!)
-- A Docker installation, which we will use to launch Grafana in a container. Alternatively, if you have another web application you'd like to protect with Teleport, you can use that instead.
-- A host where you will run the Teleport Application Service.
+- A Docker installation, which we will use to launch Grafana in a container.
+  Alternatively, if you have another web application you'd like to protect with
+  Teleport, you can use that instead.
+- A host where you will run the Teleport Application Service. If your using Grafana
+  in Docker, we include instructions to install Teleport in the container.
```

Comment:
I'm not sure we should advise this - it's generally not a good practice, and being a getting-started guide means the reader may be inexperienced or new to Teleport.

How about a docker-compose example instead?

Reply:
Not sure if you saw this, it explains my reasoning:

> While it isn't SoP to have multiple apps in a local docker container, it (sometimes) is for the App service to run alongside the app, negating the need for mutual auth between the app and agent. For the latter, this config more closely resembles a real-world scenario. Also, we don't need to confuse the reader by jumping between multiple containers.

docker-compose increases the complexity. What if I explained that my reasoning in the doc, that this is designed to resemble an "agent and app on vm" scenario.

Reply:
P.S. #24026 radically changed this guide so I'll have to review and edit these changes anyway. With an entire docker -based tab, it makes more sense to set up another container with the teleport image.

Reply:
I saw the reasoning, but I'm not sure I agree with it.

Sure, docker-compose increases the complexity, but what good is a simpler guide that instructs you to do something you'd never actually want to do in the real world?

If we want to keep it simpler and write the guide as if we're running a bunch of separate services on a single VM, then we should do just that (and remove the docker mentions) rather than treating a docker container as if it were a VM that you install extra software on.

If I'm a reader looking for a Docker guide, I'd expect to see use of Teleport's official Docker images, and patterns that align with industry best practices for how to run software in containers. I would not expect to see instructions for manually hacking Teleport into other images.